.buildkite/scripts/publish/publish-common.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+if [[ "${CURDIR:-}" == "" ]]; then
+  echo "!! CURDIR is not set. Exiting."
+  exit 2
+fi
+
+function realpath {
+  echo "$(cd "$(dirname "$1")"; pwd)"/"$(basename "$1")";
+}
+
+export SCRIPT_DIR="$CURDIR"
+export BUILDKITE_DIR=$(realpath "$(dirname "$SCRIPT_DIR")")
+export PROJECT_ROOT=$(realpath "$(dirname "$BUILDKITE_DIR")")
+
+VERSION_PATH="$PROJECT_ROOT/product_version"
+export VERSION=$(cat $VERSION_PATH)
+
+if [[ "${USE_SNAPSHOT:-}" == "true" ]]; then
```

Comment:
how does `USE_SNAPSHOT` get set in practice?

Reply:
It can be set in the env when running from buildkite. Here's a screenshot of the UI for Connectors:
![Screenshot 2024-08-23 at 10 22 15](https://github.com/user-attachments/assets/8e3347ff-b5b9-49c4-b1ea-0c5655d6bcbc)

Reply:
ah, gotcha, wasn't sure if that was how we were expecting to set the option

in that case, could it be a nicer user experience to use a formal input step, to make the existence of this snapshot option more obvious to people kicking off builds? https://buildkite.com/docs/pipelines/input-step

Reply:
I didn't know this was a thing 🤯 
I've added a step for this in this commit: https://github.com/elastic/crawler/pull/103/commits/326e36ef92b5523206f9d1655696475ab9b8e90a

Reply:
nice!